### PR TITLE
Modify all.sh to make more clear about errors during installation.

### PR DESCRIPTION
**Pull request by lzbj**
Address issue #368.

### CODE REVIEW

File: tools/ubuntu-setup/all.sh [hunk 1]
```diff
@@ -1,23 +1,92 @@
...
-source ./misc.sh
+source ./misc.sh &
+pid=$!
+wait $pid
+status=$?
+echo "$pid finished with status $status"
+if [ $status -ne 0 ]
+then
+    let ERRORS=ERRORS+1
+fi
```
Comment (rabbah) [inline tools/ubuntu-setup/all.sh]: the above can be factored into a bash function so that it's not repeated per step.

Comment (lzbj) [inline tools/ubuntu-setup/all.sh]: Yes,correct.

**Comment by rabbah** [comment]
Thanks for making the change.

I notice you have merge commits. You should use git fetch and git rebase upstream master rather than a git pull. This will allow a fast forward "merge" which preserves our linear got history. Also please squash your commits.

**Comment by lzbj** [comment]
Sorry, please ignore this request.

**Comment by lzbj** [comment]
Anyway, thank you rabbah, I'll use the command you recommended and take caution next time.

**Pull request closed without merging**